### PR TITLE
Add support for azure maps account

### PR DESCRIPTION
**Pull request by reshmarajan1730**
# [Issue-id](https://github.com/aztfmod/terraform-azurerm-caf/issues/1713)

## PR Checklist

---

- [x] I have added example(s) inside the [./examples/] folder
- [ ] I have added the example(s) to the integration test list for [normal (~30 minutes)](./workflows/standalone-scenarios.json) or [long runner >30 minutes](./workflows/standalone-scenarios-longrunners.json)
- [x] I have checked the [coding conventions as per the wiki](https://github.com/aztfmod/terraform-azurerm-caf/wiki)
- [x] I have checked to ensure there aren't other open Pull Requests for the same update/change?

## Does this introduce a breaking change

- [ ] YES
- [x] NO

## Testing
Tested standalone

### CODE REVIEW

**Comment by reshmarajan1730** [comment]
@LaurentLesle renamed to maps_accounts

**Comment by reshmarajan1730** [comment]
@arnaudlh could you please have a look

File: modules/maps/maps_account/maps_account.tf [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# naming convention
+resource "azurecaf_name" "map" {
```
Comment (arnaudlh) [inline modules/maps/maps_account/maps_account.tf:2]:
```suggestion
data "azurecaf_name" "map" {
```

File: modules/maps/maps_account/maps_account.tf [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  name          = var.settings.name
+  resource_type = "azurerm_maps_account"
+  prefixes      = var.global_settings.prefixes
+  random_length = var.global_settings.random_length
+  clean_input   = true
+  passthrough   = var.global_settings.passthrough
+  use_slug      = var.global_settings.use_slug
+}
+resource "azurerm_maps_account" "map" {
+  name                = azurecaf_name.map.result
```
Comment (arnaudlh) [inline modules/maps/maps_account/maps_account.tf:12]:
```suggestion
  name                = data.azurecaf_name.map.result
```

File: examples/variables.tf [hunk 3]
```diff
@@ -232,8 +232,8 @@ variable "mssql_mi_secondary_tdes" {
 variable "storage_accounts" {
   default = {}
 }
-variable "storage_account_file_shares" {
```
Comment (arnaudlh) [inline examples/variables.tf:235]: Do not delete other variables, just add yours :) 

```
variable "storage_account_file_shares" {
  default     = {}
}

variable "maps_accounts" {
  default     = {}
}
```

**Review by arnaudlh — CHANGES_REQUESTED** [review]
thanks for your PR @reshmarajan1730! 

please have a look at my suggestions and comments and also please add support for tags as it is support by the maps per: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/maps_account#tags

File: modules/maps/maps_account/maps_account.tf (hunk 1 above)
Comment (reshmarajan1730) [reply modules/maps/maps_account/maps_account.tf]: updated

File: modules/maps/maps_account/maps_account.tf (hunk 2 above)
Comment (reshmarajan1730) [reply modules/maps/maps_account/maps_account.tf]: updated

File: examples/variables.tf (hunk 3 above)
Comment (reshmarajan1730) [reply examples/variables.tf:235]: updated

**Comment by reshmarajan1730** [comment]
@arnaudlh I have updated the PR. With this update I have also added the feature to store the azure map account's primary and secondary access keys in a keyvault, if specified. This has also been tested.

**Review by arnaudlh — CHANGES_REQUESTED** [review]
thanks @reshmarajan1730, its looks nice and works on my environment. 

Are you able to:
1. run a ```terraform fmt``` on your files 
2. add path to your example in  ```standalone-scenarios.json``` so the CD/CI picks it. 

After that we should be ready for merge! 
Thanks.

**Review by arnaudlh — APPROVED** [review]
LGTM

**Comment by reshmarajan1730** [comment]
@arnaudlh I am not able to merge because of the branch protection

**Comment by arnaudlh** [comment]
> @arnaudlh I am not able to merge because of the branch protection

no worries, I'm the one merging ;)

**Pull request merged**